controllers/deployment_controller.go

```diff
@@ -259,6 +270,100 @@ func (r *DeploymentReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Ma
 	return nil
 }
 
+func watchRole(c controller.Controller, cl client.Client, m rbac.Matcher) error {
+	return c.Watch(&source.Kind{Type: &rbacv1.Role{}}, handler.EnqueueRequestsFromMapFunc(func(obj client.Object) []reconcile.Request {
+		var (
+			ctx         = context.Background()
+			deployments = &csibaremetalv1.DeploymentList{}
+			role        = &rbacv1.Role{}
+			ok          bool
+		)
+
+		err := cl.List(ctx, deployments)
+		if err != nil {
+			return []reconcile.Request{}
+		}
+
+		if role, ok = obj.(*rbacv1.Role); !ok {
+			return []reconcile.Request{}
+		}
+
+		if !m.MatchPolicyRules(role.Rules, []rbacv1.PolicyRule{
+			{
+				Verbs:         []string{"use"},
+				APIGroups:     []string{"security.openshift.io"},
```

Comment:
Is this check performed for all platforms?

Reply:
make check only for OCP